org.compiere.db.postgresql.provider/src/org/compiere/db/DB_PostgreSQL.java

```diff
@@ -591,105 +582,51 @@ public RowSet getRowSet (java.sql.ResultSet rs) throws SQLException
-                conn = null;
-            }
-
-            if (conn == null && exception != null)
-            {
-            	//log might cause infinite loop since it will try to acquire database connection again
-            	/*
-                log.log(Level.SEVERE, exception.toString());
-                log.fine(toString()); */
-            	System.err.println(exception.toString());
-            }
-        }
-        catch (Exception e)
-        {
-            exception = e;
-        }
+		ensureInitialized(connection);
+		
+		// If HikariCP has no available free connection this call will block until either
+		// a connection becomes available or the configured 'connectionTimeout' value is
+		// reached (after which a SQLException is thrown).
+		Connection conn = m_ds.getConnection();
 
-        try
-        {
-        	if (conn != null) {
-        		boolean trace = "true".equalsIgnoreCase(System.getProperty("org.adempiere.db.traceStatus"));
-        		int numConnections = m_ds.getNumBusyConnections();
-        		if (numConnections > 1)
-        		{
-	    			if (trace)
-	    			{
-	    				log.warning(getStatus());
-	    			}
-	    			if(numConnections >= m_maxbusyconnections && m_maxbusyconnections > 0)
-		            {
-	    				if (!trace)
-	    					log.warning(getStatus());
-		                //hengsin: make a best effort to reclaim leak connection
-		                Runtime.getRuntime().runFinalization();
-		            }
-        		}
-        	} else {
-        		//don't use log.severe here as it will try to access db again
-        		System.err.println("Failed to acquire new connection. Status=" + getStatus());
-        	}
-        }
-        catch (Exception ex)
-        {
-        }
-        if (exception != null)
-            throw exception;
-        return conn;
+		if (conn.getTransactionIsolation() != transactionIsolation)
+		{
+			conn.setTransactionIsolation(transactionIsolation);
+		}
+		if (conn.getAutoCommit() != autoCommit) 
+		{
+			conn.setAutoCommit(autoCommit);
+		}
+		
+		logDbTrace();
+
+		return conn;
 	}	//	getCachedConnection
+	
+	private final boolean dbTraceStatusEnabled = "true".equalsIgnoreCase(System.getProperty("org.adempiere.db.traceStatus")); 
+	
+	/**
+	 * Determines if the database connection status tracing feature is enabled and
+	 * logs the appropriate trace message.
+	 */
+	private void logDbTrace() {
+		if (!dbTraceStatusEnabled) {
+			return;
+		}
+		
+		log.log(Level.WARNING, getStatus());
+	}
 
-	private String getFileName ()
+	private String getPoolPropertiesFile ()
 	{
-		//
-		String base = null;
-		if (Ini.isClient())
-			base = System.getProperty("user.home");
-		else
-			base = Ini.getAdempiereHome();
+		String base = Ini.getAdempiereHome();
 		
-		if (base != null && !base.endsWith(File.separator))
+		if (base != null && !base.endsWith(File.separator)) {
 			base += File.separator;
+		}
 		
 		//
 		return base + getName() + File.separator + POOL_PROPERTIES;

```

Comment:
I think it is probably better to rename this to hikaricp.properties to avoid misreading existing pool.properties created for c3p0.

Reply:
Done.